### PR TITLE
Fix api url in default layout

### PR DESCRIPTION
**Pull request by blieb**
At the moment the api url in the default layout goes to a 404 page. Changed the link to a working url.

### CODE REVIEW

File: templates/layout/default.php [hunk 1]
```diff
@@ -42,7 +42,7 @@
         </div>
         <div class="top-nav-links">
             <a target="_blank" rel="noopener" href="https://book.cakephp.org/4/">Documentation</a>
-            <a target="_blank" rel="noopener" href="https://api.cakephp.org/4/">API</a>
+            <a target="_blank" rel="noopener" href="https://api.cakephp.org/4.1/">API</a>
```
Comment (dereuromark) [inline templates/layout/default.php:45]: Additionally, the existing URL should probably 301 to the latest.

Comment (ADmad) [inline templates/layout/default.php:45]:
```suggestion
            <a target="_blank" rel="noopener" href="https://api.cakephp.org/">API</a>
```

Homepage will direct to latest minor.

Comment (blieb) [reply templates/layout/default.php:45]: Ah yes that is a better solution then :)

Comment (blieb) [reply templates/layout/default.php]: Hmm yes. that is maybe also what I expect. I see this also don't work for /3/

File: templates/layout/default.php [hunk 2]
```diff
@@ -42,7 +42,7 @@
         </div>
         <div class="top-nav-links">
             <a target="_blank" rel="noopener" href="https://book.cakephp.org/4/">Documentation</a>
-            <a target="_blank" rel="noopener" href="https://api.cakephp.org/4/">API</a>
+            <a target="_blank" rel="noopener" href="https://api.cakephp.org/">API</a>
```
Comment (markstory) [inline templates/layout/default.php:45]: The `/4.latest/` might be a better path that redirects correctly too.

**Comment by dereuromark** [comment]
I guess for now this seems already better and working.

**Pull request merged**